File: Day9/p3/strStr.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        
        if(needle.empty() || (haystack.empty() && needle.empty()))
            return 0;
        
        if (haystack.empty())
            return -1;
        
        size_t found = haystack.find(needle);
        
        if(found != string :: npos)
            return found;
        
        return -1;
        
    }
};

File: Day9/p3/strStr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        
        if needle == '' or (haystack == '' and needle == ''):
            return 0
        
        if haystack == '' :
            return -1
          
        if needle in haystack:
            return haystack.index(needle)
        
        return -1
        
